pkg/component/runtime/manager_fake_input_test.go

```diff
@@ -3359,21 +3266,37 @@ LOOP:
 
 	// check progression, require stop fake-0 before start fake-1
 	stateProgressionWG.Wait()
-	comp0Stopped := false
+	var comp0Started, comp0Stopped, comp0Failed,
+		comp1Started, comp1Failed bool
 	for _, step := range stateProgression {
-		if step.componentID == IDComp0 &&
-			step.state.State == client.UnitStateStopped {
-			comp0Stopped = true
-		}
-		if step.componentID == IDComp1 &&
-			step.state.State == client.UnitStateStarting {
-			require.True(t, comp0Stopped)
-			break
+		switch step.componentID {
+		case IDComp0:
+			switch step.state.State {
+			case client.UnitStateStarting:
+				comp0Started = true
+			case client.UnitStateStopped:
+				comp0Stopped = true
+			case client.UnitStateDegraded, client.UnitStateFailed:
+				comp0Failed = true
+			}
+		case IDComp1:
+			switch step.state.State {
+			case client.UnitStateStarting:
+				comp1Started = true
+			case client.UnitStateDegraded, client.UnitStateFailed:
+				comp1Failed = true
+			}
```

Comment:
I'm not convinced you can ignore the order of operations.

If you don't do this, you risk bringing back a pervious bug. See https://github.com/elastic/elastic-agent/issues/4286#issuecomment-1954818251 and the description of the change that introduced this test: https://github.com/elastic/elastic-agent/pull/2226

> Currently we start new components before stopping old. This is usually fine unless component is a service (start/stop is slower).
When new service is started old one can be still running and using resources needed for new service to start.
We can than see something like bind port already in use.

We care about the order of operations for the same input type, otherwise we can create situations where the inputs can't start.

Reply:
> I'm not convinced you can ignore the order of operations.
> 
> If you don't do this, you risk bringing back a previous bug. See [#4286 (comment)](https://github.com/elastic/elastic-agent/issues/4286#issuecomment-1954818251) and the description of the change that introduced this test: #2226
> 
> > Currently we start new components before stopping old. This is usually fine unless component is a service (start/stop is slower).
> > When new service is started old one can be still running and using resources needed for new service to start.
> > We can than see something like bind port already in use.
> 
> We care about the order of operations for the same input type, otherwise we can create situations where the inputs can't start.

done, I kept the check on the order of the events